crates/control/src/game_state_filter.rs

```diff
@@ -77,11 +77,27 @@ impl GameStateFilter {
             })
             .unwrap_or(false);
 
+        let ball_detected_far_from_penalty_spot = context
+            .ball_position
+            .map(|ball| {
+                let absolute_ball_position = *context.robot_to_field * ball.position;
+                distance(
+                    &absolute_ball_position,
+                    &Point2::new(
+                        -context.field_dimensions.length / 2.0
+                            + context.field_dimensions.penalty_marker_distance,
+                        0.0,
+                    ),
```

Comment:
extract this to a variable to increase code readability

Reply:
extracted